pkg/extensions/tbtc/liquidation_recovery.go

```diff
@@ -0,0 +1,282 @@
+package tbtc
+
+import (
+	cecdsa "crypto/ecdsa"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcutil/hdkeychain"
+	"github.com/keep-network/keep-ecdsa/pkg/ecdsa"
+)
+
+// PublicKeyToP2WPKHScriptCode converts a public key to a Bitcion p2wpkh
+// witness scriptCode that can spend an output sent to that public key's
+// corresponding address.
+//
+// [BIP143]: https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki
+func PublicKeyToP2WPKHScriptCode(
+	publicKey *cecdsa.PublicKey,
+	chainParams *chaincfg.Params,
+) ([]byte, error) {
+	// ecdsa.PublicKey and btcec.PublicKey are both method attachments to
+	// Go's crypto/ecdsa.PublicKey, so we can cast.
+	publicKeyBytes := (*btcec.PublicKey)(publicKey).SerializeCompressed()
+	// Note that the scriptCode for a p2wpkh address is the equivalent of the
+	// p2pkh scriptPubKey.
+	pubKeyAddress, err := btcutil.NewAddressPubKey(publicKeyBytes, chainParams)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"error deriving p2wpkh scriptCode from public key: [%s]",
+			err,
+		)
+	}
+	pkhAddress := pubKeyAddress.AddressPubKeyHash()
+
+	script, err := txscript.PayToAddrScript(pkhAddress)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"error deriving p2wpkh scriptCode from public key: [%s]",
+			err,
+		)
+	}
+	if len(script) > 255 {
```

Comment:
This is a great point. This check was to avoid an overflow of the `byte` cast below, but since we're specifically aiming for 0x19 as the first byte, checking that `len(script) == 25` and adjusting the error to `unexpected scriptCode length` makes sense to me.